.github/workflows/main.yml

```diff
@@ -49,6 +51,9 @@ jobs:
       - name: Run rustfmt
         run: cargo fmt -- --check
 
+      - name: Run clippy
+        run: cargo clippy --workspace --all-targets --all-features -- -D warnings
```

Comment:
this will imply that contributions to the codebase must now pass clippy, and no warnings allowed.
I see that a clippy lint is enforcing two rules:
- `#![allow(clippy::enum_variant_names)]`
- `#[allow(clippy::type_complexity)]`

Reply:
> this will imply that contributions to the codebase must now pass clippy

yes. I find clippy is a good teacher for new contributors. They learn about idiomatic rust 👍

> and no warnings allowed

I think that all clippy lints are treated as warnings, so if you don't add `-D warnings` then this step will pass 😅



> What happens if clippy "stabilizes" and adds a new default?

clippy adds new defaults on every rust release more or less. We pin the rust version in CI, so whenever we bump the rust version we need to fix new clippy lints, too (if any)